src/components/mainContent/PostsList.js

```diff
@@ -1,22 +1,80 @@
 import React from "react";
 import Post from "../posts/Post";
 import CommentSection from "../comments/CommentSection";
-
+import ForwardButton from "./ForwardButton";
+import "./mainContent.css";
 export default function PostsList(props) {
-  /*This function takes the list of post/followups from our backend and renders it accordingly
-Sections can either be posts or followups (Interactive sections might be seperate in future as well*/
-  return (
-    <div>
-      {props.content.map((section, index) => {
-        /*Not using ternaries since there might be 3 possibilities if we make
-          interactive sections seperate from posts in the future
-          */
-        if (section.post)
-          return <Post key={index} postContent={section.post} />;
+  
+
+  /*This function takes the list of post/followups, takes all the previousSections that are already loaded,
+  and makes them show all their text*/
+  let completedContent = props.content
+    .slice(0, props.visibleSections)
+    .map((section, index) => {
+      /*Not using ternaries since there might be 3 possibilities if we make
+      interactive sections seperate from posts in the future
+      */
+      if (section.post)
+      {
+        return (
+          <Post
+            key={index}
+            postContent={section.post}
+            currentVisibleText={section.post.bodyText.length}
+          />
+        );
+      }
+      else {
         if (section.followups)
           return <CommentSection key={index} followups={section.followups} />;
-        return "";
-      })}
+      return "";
+      }
+    });
+  /* This function takes the section that the user is currently on, and makes it so that its content
+    can change depending on how the DirectionButtons are pressed*/
+  let currentContent = props.content
+    .slice(props.visibleSections, props.visibleSections + 1)
+    .map((section, index) => {
+      /*Not using ternaries since there might be 3 possibilities if we make
+    interactive sections seperate from posts in the future
+    */
+      if (section.post)
+        return (
+          <Post
+            key={index}
+            postContent={section.post}
+            currentVisibleText={props.currentVisibleText}
+            setCurrentVisibleText={props.setCurrentVisibleText}
+          />
+        );
+      if (section.followups)
+        return (
+          <CommentSection
+            key={index}
+            followups={section.followups}
+            currentVisibleText={props.currentVisibleText}
+            setCurrentVisibleText={props.setCurrentVisibleText}
+          />
+        );
+      return "";
```

Comment:
This looks like it should be an `if-else-if` chain, and it would definitely help to add braces for readability; especially since your return actually spans multiple lines.